jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt

```diff
@@ -45,10 +41,26 @@ class PollMessagePaneTest : BaseCoroutineTest() {
             tableModel.waitForModelToBeAtLeast(1)
         }
 
-        assertThat(tableModel.items.first().messageId()).isEqualTo("XYZ")
-        assertThat(tableModel.items.first().body()).isEqualTo("ABC")
-        assertThat(tableModel.items.first().attributes().getValue(MessageSystemAttributeName.SENDER_ID)).isEqualTo("1234567890:test1")
-        assertThat(tableModel.items.first().attributes().getValue(MessageSystemAttributeName.SENT_TIMESTAMP)).isEqualTo("111111111")
+        assertThat(tableModel.items.first()).isEqualTo(message)
+    }
+
+    @Test
+    fun `Messages are de-duplicated by id`() {
+        val client = mockClientManagerRule.create<SqsClient>()
+        val message = aMessage()
+        val anotherMessage = aMessage()
+        whenever(client.receiveMessage(Mockito.any<ReceiveMessageRequest>())).thenReturn(
+            ReceiveMessageResponse.builder().messages(message, message, anotherMessage).build()
+        )
+        val pane = PollMessagePane(projectRule.project, client, queue)
+        val tableModel = pane.messagesTable.tableModel
+        runBlocking {
+            pane.requestMessages()
+            tableModel.waitForModelToBeAtLeast(1)
+        }
+
+        assertThat(tableModel.items).hasSize(2)
```

Comment:
This should be 1 if we are testing de-dupe

Reply:
there's two different messages in the list `message` (x 2) and `anotherMessage`